cmd/team-member-updater/main.go

```diff
@@ -0,0 +1,283 @@
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+	"sigs.k8s.io/yaml"
+)
+
+type options struct {
+	membersPath string
+	reposPath   string
+	configPath  string
+	logLevel    string
+}
+
+func (o *options) Validate() error {
+	if o.membersPath == "" {
+		return errors.New("required flag --members was unset")
+	}
+
+	if o.reposPath == "" {
+		return errors.New("required flag --repos was unset")
+	}
+
+	if o.configPath == "" {
+		return errors.New("required flag --config was unset")
+	}
+
+	level, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %v", err)
+	}
+	logrus.SetLevel(level)
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.membersPath, "members", "", "Path to AOS Team Member Tracking spreadsheet.")
+	fs.StringVar(&o.reposPath, "repos", "", "Path to AOS Repository Tracking spreadsheet.")
+	fs.StringVar(&o.configPath, "config", "", "Path to peribolos config to update.")
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	var orgConfig struct {
+		Orgs map[string]org.Config `json:"orgs,omitempty"`
+	}
+	rawConfig, err := ioutil.ReadFile(o.configPath)
```

Comment:
As someone who doesn't initially know how the program should operate, it's hard to keep track of this number of local variables. E.g. none of the `raw*` or `*Records` variable needs to exist a few lines after it is created. Can we break `main` into a few more tightly-scoped functions?